src/leds.c

```diff
@@ -322,6 +323,32 @@ static void anim_felony(Leds *leds, const LedStrip *strip, const LedBar *bar, fl
     }
 }
 
+static void anim_rainbow_cycle(Leds *leds, const LedStrip *strip, float time) {
+    uint8_t count = 10;
+    uint8_t segment = 256 / count;
+    uint8_t color_idx = ((uint8_t) (time / 0.1f) % count) * segment;
+    strip_set_color(leds, strip, hue_to_color(color_idx), strip->brightness, 1.0f);
+}
+
+static void anim_rgb_fade(Leds *leds, const LedStrip *strip, float time) {
+    uint8_t offset = floorf(fmodf(time, 4.0f) * 255.0f);
```

Comment:
`floorf` call is redundant since the float is implicitly converted during assignment, which does the same thing.

You're multiplying by 255 so the `fmodf` can be by `1.0` as this way you're overflowing the `uint8_t` up to three times and the result is the same.